src/DynamoCoreWpf/ViewModels/Search/BrowserItemViewModel.cs

```diff
@@ -430,24 +421,46 @@ private void RemoveFromItems(IEnumerable<ISearchEntryViewModel> oldItems)
 
         private void AddToItems(IEnumerable<ISearchEntryViewModel> newItems)
         {
+            bool hasClasses = Items.FirstOrDefault() is ClassesNodeCategoryViewModel;
+
             foreach (var entry in newItems)
             {
-                var first = Items.Select((x, i) => new { x.Name, Idx = i })
-                    .FirstOrDefault(
-                        x =>
-                            string.Compare(
-                                x.Name,
-                                entry.Name,
-                                StringComparison.Ordinal)
-                                >= 0);
+                var nextLargerItemIndex = -1;
+                foreach (var item in Items.Where(cat => !(cat is ClassesNodeCategoryViewModel)))
+                {
+                    if (string.Compare(item.Name, entry.Name, StringComparison.Ordinal) >= 0)
+                    {
+                        nextLargerItemIndex = Items.IndexOf(item);
+                        break;
+                    }
+                }
                 // Classes must be first in any case.
                 if (entry is ClassesNodeCategoryViewModel)
```

Comment:
Let's move this `if (entry is ClassesNodeCategoryViewModel)` to before `var nextLargerItemIndex` because we don't want to waste time searching and not using it.